docs/excel/excel-data-types-linked-entity-cell-values.md

```diff
@@ -0,0 +1,539 @@
+  console.log("Refresh error: " + eventArgs.errors);
+  return null;
+}
+```
+
+## Error handling in linked entity load service
+
+When Excel calls your add-in to get data for a linked entity cell value, it's possible an error can occur. If Excel is unable to connect to your add-in at all (such as when it is not loaded) it will display `#CONNECT!` to the user.
+
+If your linked entity load service function encounters an error, it should throw an error. The error will cause Excel to show `#CONNECT!` to the user.
+
+The following code shows how to handle an error in a linked entity load service function.
+
+```javascript
+async function productLinkedEntityService(request: any): Promise<any> {
+    const notAvailableError = new CustomFunctions.Error(CustomFunctions.ErrorCode.notAvailable);
+    try {
+      // Create an return a new linked entity cell value.
+        let linkedEntityResult = ...
+      ...
+        if (!linkedEntityResult) {
+        // Throw an error to signify to Excel that resolution/refresh of the requested linkedEntityId failed.
+        throw notAvailableError;
+      }
+      ...
+    } catch (error) {
+        console.error(error);
+        throw notAvailableError;
+    }
+}
+```
+
+## Debugging the linked entity load service
+
+Most add-in functionality for linked entity data types can be debugged using the guidance in [Overview of debugging Office Add-ins](../testing/debug-add-ins-overview.md). However, the linked entity load service function can be implemented in a shared runtime, or a JavaScript-only runtime (also know as a custom function runtime.) If you choose to implement the function in a JavaScript-only runtime, use the debugging guidance in JavaScript-only runtime for [Custom functions debugging in a non-shared runtime](custom-functions-debugging.md).
+
+The linked entity load service function uses the custom function architecture, regardless of which runtime you use. However, there are significant differences from regular custom functions.
+Differences:
+
+- They will not appear to end users for usage in formulas.
+- They do not support the JSDoc tags `@streaming` or `@volatile`.
+
+Similarities:
+
+- They use [Custom functions naming and localization](custom-functions-naming.md).
+- They use the same error handling approach.
+
+## Behavior in Excel 2019 and earlier
+
+If someone opens a worksheet with linked entity cell values on an older version of Excel that doesn’t support linked entity cell values, Excel will show the cell values as errors. This is the designed behavior. This is also why you set the `basicType` to `Error` and the `basicValue` to `#VALUE!` every time you insert or update a linked entity cell value. This is the error that Excel will fall back to on older versions.
+
+## Best practices
+
+- Register linked entity data domains in the initialization code `Office.OnReady`, so that the user will have immediate functionality such as refreshing the linked entity cell values.
```

Comment:
```suggestion
- Register linked entity data domains in the initialization code `Office.OnReady` so that the user has immediate functionality and refreshed linked entity cell values.
```

Reply:
fixed